08 update window on new packets

mayurhyphenOs opened the pull request.

gabhijit commented on src/components/Table.js:
```diff
@@ -81,12 +81,30 @@ const Table = ({ getSelectedPacket, packets, config }) => {
           setWindowStart(frameno);
           setWindowEnd(frameno + config.packetWindowSize);
         }
+        updateWindow(frameno);
```
This call will happen on every packet.  This is not required. Take out the `frameno` out of the `for` loop initialize it to `-Infinity` so that it will never be greater than `windowEnd` if we don't enter the `if(packet)` condition in the for loop and then after the `for` loop is completed, call `maybeUpdateWindow` (or `updateWindow` for now.)

gabhijit commented on src/components/Table.js:
```diff
@@ -81,12 +81,30 @@ const Table = ({ getSelectedPacket, packets, config }) => {
+   * new packets after windowEnd are greater than or equal to config.jumpSize (sizable packets after window)
+   * @param {number} frameno - frame number of received packet
+   */
+  const updateWindow = (frameno) => {
```
Call it `mayBeUpdateWindow` already and return `true/false` or may be `didWindowUpdate`?

mayurhyphenOs commented:
# Requirements:

1. Update window and scroll to the last element only if scroll position is at the end.
2. Pause auto-scroll and window update if the scroll is not at the end position.
3. if scroll position is at begin, jump up by config.jumpSize.

Cases:
  1. Initial scroll position (no scroll at all): should not update window or scroll position
  2. Changed scroll position (scrolled but not reached the end): Should not update window or scroll position
  3.  Changed scroll position to end (scrolled and reached end): Should update window and scroll to the end only on new data reception.
  4. Changed scroll position to end (scrolled and reached end) but no new data: Should not update and scroll to the last automatically. User will have to iterate through each window to reach the last received data.
  5. Changed scoll position to end and moved up slightly (scrolled to end and reverted to position which is not end position): Should not update window and scroll to last position

# Implementation Approach:
1. set scrollEnd position in state and use it to invoke updateWindow() on new data reception.

We will use `isScrollEnd` from `handleScroll()` to set `atScrollEnd` state. This will update window and scroll to last table row only if current scroll position is at end 

```

    if (atScrollEnd) 
      if (didUpdateWindow(frameno)) 
        tableLastRowRef.current.scrollIntoView({behavior: "smooth"}) // Scrolls until last table row is in view 

```

mayurhyphenOs commented:
# Final Approach

## Requirements:

1. Autoscroll updates the window and scrolls to last table row on new data reception.
2. Autoscroll button will provide autoscroll status and can also toggle it `on` or `off`.
3. Autoscroll will be `on` by default.
4. Autoscroll will be turned `off` whenever the onScroll event (`handleScroll()`) of table container is triggered.

Implementation Approach:

1. Use `autoscroll` state (initial: true) to toggle autoscroll `on` and `off` and also show its current status.
2.  On each new data reception, if autoscroll is `on`, update the window and scroll last table row into view if window updated.
3. if user triggers onScroll event by scrolling the table, autoscroll will be turned `off`.
